### PR TITLE
APL-1647 Add i18n links to the mobile stores

### PR DESCRIPTION
**Pull request by robvk**
Decided to put it here so that services other than the email service can also use it if needed in the future.

### CODE REVIEW

File: v2/i18n/icons.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	language.EuropeanPortuguese:   generateGooglePlayStoreIconLink("pt"),    // pt-PT
+	language.Russian:              generateGooglePlayStoreIconLink("ru"),    // ru
+	language.Swedish:              generateGooglePlayStoreIconLink("sv"),    // sv
+	language.Thai:                 generateGooglePlayStoreIconLink("th"),    // th
+	language.Turkish:              generateGooglePlayStoreIconLink("tr"),    // tr
+	language.Ukrainian:            generateGooglePlayStoreIconLink("uk"),    // uk
+	language.Vietnamese:           generateGooglePlayStoreIconLink("vi"),    // vi
+	language.Chinese:              generateGooglePlayStoreIconLink("zh"),    // zh
+	language.SimplifiedChinese:    generateGooglePlayStoreIconLink("zh"),    // zh-Hans
+	language.TraditionalChinese:   generateGooglePlayStoreIconLink("zh"),    // zh-Hant
```
Comment (sid-ag5) [inline v2/i18n/icons.go]: If you want to get the code a bit smaller, I'd specify the mapping once from `language.Tag` to code (e.g. "zh"), assuming that this mapping is always the same. And then use that in both of the icon functions (then you also only have to call `generateGooglePlaystoreIconLink` once)

File: v2/i18n/icons_test.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		tag:                 language.SimplifiedChinese, // zh-Hans
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/zh.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/zh.png",
+	},
+	{
+		tag:                 language.TraditionalChinese, // zh-Hant
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/zh.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/zh.png",
+	},
+}
```
Comment (sid-ag5) [inline v2/i18n/icons_test.go:181]: I'm not sure if testing a mapping like this is the way to go (also from a maintainability point of view: if we're going to have the full mapping hardcoded in the codebase like this anyway and we're more sure of this test code than the code in the actual logic, then why not just use this as the main logic and not test it :P).

I'd focus on one basic test for the main part of the URL, and then edge cases like there not being any language tags or the language tags not being in the mapping

**Review by sid-ag5 — COMMENTED** [review]
Added some maintainability comments

File: v2/i18n/icons.go (hunk 1 above)
Comment (robvk) [reply v2/i18n/icons.go]: Good point, done.

File: v2/i18n/icons_test.go (hunk 2 above)
Comment (robvk) [reply v2/i18n/icons_test.go:181]: That test was more a 'run it' kind of thing to see if my algorithm was ok when I didn't know go had maps and I tried to make it all dynamic based on the language! But that became a mess with the exceptions and differences so decided to hard code it afterwards which makes it a duplicate indeed.

Added better tests now, but think it is ok to leave it in for now if someone later wants to make an algorithm for it and wants to check if it still gives the same result.

File: v2/i18n/icons.go [hunk 3]
```diff
@@ -64,39 +64,39 @@ func generateGooglePlayStoreIconLink(lang string) string {
...
+	language.EuropeanPortuguese:   "pt",    // pt-PT
+	language.Russian:              "ru",    // ru
+	language.Swedish:              "sv",    // sv
+	language.Thai:                 "th",    // th
+	language.Turkish:              "tr",    // tr
+	language.Ukrainian:            "uk",    // uk
+	language.Vietnamese:           "vi",    // vi
+	language.Chinese:              "zh",    // zh
+	language.SimplifiedChinese:    "zh",    // zh-Hans
+	language.TraditionalChinese:   "zh",    // zh-Hant
```
Comment (sid-ag5) [inline v2/i18n/icons.go]: These two mappings should now be the same (a general languageTagToCode mapping), so one of them can be removed

Comment (robvk) [reply v2/i18n/icons.go]: Ah I see what you mean. Apple is weird and does se for Swedish which meant they aren't the exact same, but we can fix that on the icon side of things. Does feel nicer to have it like this

**Review by sid-ag5 — APPROVED** [review]
